File: MeetHut.Backend/ClientApp/src/app/web/dtos/room.dto.ts
export class RoomPublicDTO {
  name: string;
  publicId: string;

  constructor(name: string, publicId: string) {
    this.name = name;
    this.publicId = publicId;
  }
}

export class RoomDTO extends RoomPublicDTO {
  id: number;
  creation: Date;
  lastUpdate: Date;
  name: string;
  publicId: string;
  startTime: string;
  endTime: string;
  owner: {
    id: number;
    userName: string;
  };
  participants: number;
  onlineParticipants: number;

  constructor(
    id: number,
    creation: Date,
    lastUpdate: Date,
    name: string,
    publicId: string,
    startTime: string,
    endTime: string,
    owner: any,
    participants: number,
    onlineParticipants: number
  ) {
    super(name, publicId);
    this.id = id;
    this.creation = creation;
    this.lastUpdate = lastUpdate;
    this.name = name;
    this.publicId = publicId;
    this.startTime = startTime;
    this.endTime = endTime;
    this.owner = owner;
    this.participants = participants;
    this.onlineParticipants = onlineParticipants;
  }
}

File: MeetHut.Services/Meet/DTOs/RoomDTO.cs
using System;
using MeetHut.Services.Application.DTOs;

namespace MeetHut.Services.Meet.DTOs
{
    /// <summary>
    /// Room DTO
    /// </summary>
    public class RoomDTO
    {
        /// <summary>
        /// Id
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Creation
        /// </summary>
        public DateTime Creation { get; set; }

        /// <summary>
        /// Last update
        /// </summary>
        public DateTime LastUpdate { get; set; }

        /// <summary>
        /// Room name
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Public Id
        /// </summary>
        public string PublicId { get; set; }

        /// <summary>
        /// Owner Id
        /// </summary>
        public int OwnerId { get; set; }

        /// <summary>
        /// Owner
        /// </summary>
        public UserDTO Owner { get; set; }

        /// <summary>
        /// Start time
        /// </summary>
        public DateTime? StartTime { get; set; }

        /// <summary>
        /// End time
        /// </summary>
        public DateTime? EndTime { get; set; }

        /// <summary>
        /// Room is locked
        /// </summary>
        public bool Locked { get; set; } = false;

        /// <summary>
        /// Number of added participants
        /// </summary>
        public int? Participants { get; set; }

        /// <summary>
        /// Number of online (livekit reportedly) users
        /// </summary>
        public int? OnlineParticipants { get; set; }
    }
}